main.cpp

```diff
@@ -40,13 +40,155 @@
 #include <modules/clightning/module.h>
 #endif
 
-#ifdef CUSTOM_MUTATOR_BOLT11
-#include <modules/bolt11mutator/bech32.h>
+#if defined(CUSTOM_MUTATOR_BOLT11) || defined(CUSTOM_MUTATOR_BOLT12_OFFER)
+#include <modules/custommutator/bech32.h>
+#include <modules/custommutator/customcrossover.h>
 #endif
 
+#ifdef CUSTOM_MUTATOR_BOLT12_OFFER
+constexpr std::string_view bech32_hrp = "lno";
+constexpr std::string_view dummy_initial_input = "lno1";
+#endif
 
 std::shared_ptr<bitcoinfuzz::Driver> driver = nullptr;
 
+#ifdef CUSTOM_MUTATOR_BOLT12_OFFER
+// A custom mutator that decodes the bech32 input, mutates the decoded input,
+// and then re-encodes the mutated input. This produces an input corpus that
+// consists entirely of correctly encoded bech32 strings, enabling efficient
+// fuzzing of the bolt12 decoding logic without the fuzzer getting stuck on
+// fuzzing the bech32 decoding logic. This custom mutator is originally from
+// core-lightning:
+// https://github.com/ElementsProject/lightning/blob/release-v25.02.2/tests/fuzz/bolt12.h#L54
+extern "C" size_t LLVMFuzzerMutate(uint8_t *Data, size_t Size, size_t MaxSize);
+extern "C" size_t LLVMFuzzerCustomMutator(uint8_t *fuzz_data, size_t size, size_t max_size,
+                                          unsigned int seed);
+extern "C" size_t LLVMFuzzerCustomCrossOver(const uint8_t *in1, size_t in1_size, const uint8_t *in2,
+                                            size_t in2_size, uint8_t *out, size_t max_out_size,
+                                            unsigned seed);
+
+/* Encodes a dummy bolt12 offer/invoice-request/invoice into fuzz_data and
+ * returns the size of the encoded string. */
+static size_t initial_input(uint8_t *fuzz_data, size_t max_size)
+{
+    size_t output_size = std::min(max_size, dummy_initial_input.size());
+    std::memcpy(fuzz_data, dummy_initial_input.data(), output_size);
+
+    return output_size;
+}
+
+/* A custom mutator that decodes the bech32 input, mutates the decoded input,
+ * and then re-encodes the mutated input. This produces an input corpus that
+ * consists entirely of correctly encoded bech32 strings, enabling efficient
+ * fuzzing of the bolt12 decoding logic without the fuzzer getting stuck on
+ * fuzzing the bech32 decoding logic. */
+extern "C" size_t LLVMFuzzerCustomMutator(uint8_t *fuzz_data, size_t size,
+                                          size_t max_size, unsigned int seed)
+{
+    std::string input_str(reinterpret_cast<char *>(fuzz_data), size);
```

Comment:
I think `input_str` could be a `string_view`, and `DecodeNoChecksum` could take a `string_view` param, which would avoid another copy.